File: LeetCode/LeetCode Practice Problems/1436- Destination City.cpp
class Solution {
public:
    string destCity(vector<vector<string>>& paths) {
        map<string , string> m;
        for(auto i : paths)
        {
            m[i[0]] = i[1];
            if(m.find(i[1]) == m.end())
                m[i[1]] = "";
        }
        
        string ans = "";
        for(auto i : m)
            if(i.second == "")
                ans = i.first;
        
        return ans;
    }
};

File: LeetCode/LeetCode Practice Problems/1436-Destination City.java
class Solution {
    public String destCity(List<List<String>> paths) {
        HashSet<String> src = new HashSet();
        HashSet<String> set = new HashSet();
        
        for(List<String> path: paths) {
            src.add(path.get(0));
            set.add(path.get(1));
        }
        
        String ans = "";
        
        for(String key: set) {
            if(!src.contains(key)) {
                ans = key;
                break;
            }
        }
        
        return ans;
    }
}
